[master] Upgrade to latest dependencies

knative-automation opened the pull request:
Justification, Check new logs by Dave -slinkydeveloper

/cc we don't use Prow
/assign we don't use Prow

Produced via:
```shell
./hack/update-deps.sh --upgrade --release v0.21
./hack/update-codegen.sh
```

Details:
```
bumping knative.dev/eventing a414aee...1897543:
  > 1897543 [# 4717] Update e2e broker defaults webhook test (# 4720)
  > c5b9eeb Unify usage of YAML libraries (# 4667)
  > 2e96fd0 Now the config-imc-event-dispatcher values are injected to the dispatcher through envs (# 4543)
  > 05a3a27 note that these only pertain to this repo (# 4733)
  > 21d91e4 Format markdown (# 4734)
  > 47d9937 Update spec.md with v1 API (# 4594)
  > 343dec3 Use the v1 api package... (# 4727)
  > ff5109d re-submit # 4707 (# 4730)
  > b1882d6 [# 4721] Update e2e broker default webhook test (# 4722)
  > 7d12333 Disabling servicelinks in all the components (# 4714)
bumping knative.dev/pkg 51c72e2...acbf2af:
  > acbf2af Harden logstream against non-object events (# 1984)
  > 93874f0 Drainer supports customized kubelet probe check (# 1977)
bumping knative.dev/hack 4ed1ecb...d946d25:
  > d946d25 Revert "Add hermetic go mod aware handling to run_go_tool (# 38)" (# 39)
  > ea9c9a0 Add hermetic go mod aware handling to run_go_tool (# 38)
```

dprotaso commented:
There's a bug in the `modlog` output 

I thought all pseudo-versions had a  `v0.0.0` prefix but I was wrong

dprotaso commented:
/hold

knative-automation committed:
upgrade to latest dependencies

bumping knative.dev/eventing a414aee...1897543:
  > 1897543 [# 4717] Update e2e broker defaults webhook test (# 4720)
  > c5b9eeb Unify usage of YAML libraries (# 4667)
  > 2e96fd0 Now the config-imc-event-dispatcher values are injected to the dispatcher through envs (# 4543)
  > 05a3a27 note that these only pertain to this repo (# 4733)
  > 21d91e4 Format markdown (# 4734)
  > 47d9937 Update spec.md with v1 API (# 4594)
  > 343dec3 Use the v1 api package... (# 4727)
  > ff5109d re-submit # 4707 (# 4730)
  > b1882d6 [# 4721] Update e2e broker default webhook test (# 4722)
  > 7d12333 Disabling servicelinks in all the components (# 4714)
bumping knative.dev/pkg 51c72e2...acbf2af:
  > acbf2af Harden logstream against non-object events (# 1984)
  > 93874f0 Drainer supports customized kubelet probe check (# 1977)
bumping knative.dev/hack 4ed1ecb...d946d25:
  > d946d25 Revert "Add hermetic go mod aware handling to run_go_tool (# 38)" (# 39)
  > ea9c9a0 Add hermetic go mod aware handling to run_go_tool (# 38)

Signed-off-by: Knative Automation <automation@knative.team>

dprotaso commented:
Ok looks good
